sherpa/__init__.py

```diff
@@ -63,9 +63,447 @@ def format(self, record):
+    booktitle = {Astronomical Data Analysis},
+         year = 2001,
+       editor = {{Starck}, Jean-Luc and {Murtagh}, Fionn D.},
+       series = {Society of Photo-Optical Instrumentation Engineers (SPIE) Conference Series},
+       volume = {4477},
+        month = nov,
+        pages = {76-87},
+          doi = {10.1117/12.447161},
+archivePrefix = {arXiv},
+       eprint = {astro-ph/0108426},
+ primaryClass = {astro-ph},
+       adsurl = {https://ui.adsabs.harvard.edu/abs/2001SPIE.4477...76F},
+      adsnote = {Provided by the SAO/NASA Astrophysics Data System}
+}
+
+@INPROCEEDINGS{2007ASPC..376..543D,
+   author = {{Doe}, S. and {Nguyen}, D. and {Stawarz}, C. and {Refsdal}, B. and
+	{Siemiginowska}, A. and {Burke}, D. and {Evans}, I. and {Evans}, J. and
+	{McDowell}, J. and {Houck}, J. and {Nowak}, M.},
+    title = "{Developing Sherpa with Python}",
+booktitle = {Astronomical Data Analysis Software and Systems XVI},
+     year = 2007,
+   series = {Astronomical Society of the Pacific Conference Series},
+   volume = 376,
+   editor = {{Shaw}, R.~A. and {Hill}, F. and {Bell}, D.~J.},
+    month = oct,
+    pages = {543},
+   adsurl = {http://adsabs.harvard.edu/abs/2007ASPC..376..543D},
+  adsnote = {Provided by the SAO/NASA Astrophysics Data System}
+}
+"""
+
+
+def _make_citation(version, title, date, authors, idval, latest=True):
+
+    year = date.year
+    month = date.strftime('%b').lower()
+    nicedate = date.strftime('%B %d, %Y')
+
+    authnames = ' and\n                  '.join(authors)
+
+    if latest:
+        out = "The latest release of Sherpa is {}\n".format(title)
+        out += "released on {}.".format(nicedate)
+    else:
+        out = "Sherpa {} was released on {}.".format(version, nicedate)
+
+    out += """
+
+@software{{sherpa_{year}_{idval},
+  author       = {{{authnames}}},
+  title        = {{{title}}},
+  month        = {month},
+  year         = {year},
+  publisher    = {{Zenodo}},
+  version      = {{{version}}},
+  doi          = {{10.5281/zenodo.{idval}}},
+  url          = {{https://doi.org/10.5281/zenodo.{idval}}}
+}}
+
+""".format(authnames=authnames, idval=idval, title=title,
+           version=version, year=year, month=month)
+
+    out += DEFAULT_CITATION
+    return out
+
+
+def _get_citation_hardcoded(version):
+    """Retrieve the citation information.
+
+    Parameters
+    ----------
+    version : str
+        The version to retrieve the citation before. It is expected
+        to be '4.8.0' or higher.
+
+    Returns
+    -------
+    citation : str or None
+        Citation information if known, otherwise None.
+
+    """
+
+    def todate(year, mnum, dnum):
+        return datetime.datetime.fromisoformat('{}-{:02d}-{:02d}'.format(year, mnum, dnum))
+
+    db = 'DougBurke'
+    ol = 'Omar Laurino'
+    dn = 'dtnguyen2'
+    ta = 'Tom Aldcroft'
+    an = 'Aneta Siemiginowska'
+
+    jb = 'Jamie Budynkiewicz'
+    cd = 'Christoph Deil'
+    bs = 'Brigitta Sipocz'
+
+    wm = 'wmclaugh'
```

Comment:
If you're able to give everyone a variable then you could give most of these folks real names instead of handles, right?

Reply:
All I'm trying to do here is match the Zenodo output. While I agree that citing 'and Todd' is not the most informative thing to do, I personally feel that that's outside this particular PR. Or my paygrade.